### ISSUE TITLE
Link component with HistoryLocation encodes path twice 

### ISSUE THREAD

**Issue by WebApelsin** [rackt/react-router#595]
Suppose we use HistoryLocation and have a Link to some route with queryString:
```jsx
<Link to="route_name" query={{ keyword: "Lorem ipsum" }} />
```
The resulting path that will be passed to `window.history` object will be double-encoded:
```
http://www.anyhost.com/route_path?keyword=Lorem%2520ipsum
```
This is because the path passed to `HistoryLocation.push()` method is already encoded, but HistoryLocation repeatedly encodes it.

**Comment by artemkozlov** [comment]
I have the same experience in Chrome as well, Version 38.0.2125.111 (64-bit):
this.transitionTo('route_name', {}, { keyword: "Lorem ipsum"});

As result keyword will be double encoded.

**Comment by mjackson** [comment]
This should be fixed by db2607d4d33322acb2c77c8906068015f2254c12. Please feel free to re-open if not.
